spec/game_spec.rb

```diff
@@ -0,0 +1,336 @@
+require './game.rb'
+require './player.rb'
+
+RSpec.describe Game do
+
+    before Game do 
+        @player1 = Player.new('John','X')
+        @player2 = Player.new('Doe','O')
+        @game = Game.new(@player1,@player2) 
+    end
+           
+
+    describe '#set_board_value' do 
+    
+
+        it 'Adds X to [0,0] position of the board' do
+           
+            @game.current_player_move = @player1
+            @game.set_board_value("Q")
+            game_array = @game.board_array 
+
+            expect(game_array[0][0]).to eq(@game.current_player_move.mark)
+        end
+        
+        it'Adds O to [0,2] position of the board' do
+            @game.current_player_move = @player2
+            
+            @game.set_board_value("E")
+            
+            game_array = @game.board_array
+
+            expect(game_array[0][2]).to eq(@game.current_player_move.mark)
+        end
+        # TODO add more tests 
+        it'Adds X to [1,1] position of the board' do
+            @game.current_player_move = @player1
+            
+            @game.set_board_value("S")
+            
+            game_array = @game.board_array
+
+            expect(game_array[1][1]).to eq(@game.current_player_move.mark)
+        end
+        it'Adds O to [2,2] position of the board' do
+            @game.current_player_move = @player2
+            
+            @game.set_board_value("C")
+            
+            game_array = @game.board_array
+
+            expect(game_array[2][2]).to eq(@game.current_player_move.mark)
+        end
+
+        it'Adds O to [1,2] position of the board' do
+            @game.current_player_move = @player2
+            
+            @game.set_board_value("D")
+            
+            game_array = @game.board_array
+
+            expect(game_array[1][2]).to eq(@game.current_player_move.mark)
+        end
+        it'Adds X to [2,0] position of the board' do
+            @game.current_player_move = @player1
+            
+            @game.set_board_value("Z")
+            
+            game_array = @game.board_array
+
+            expect(game_array[2][0]).to eq(@game.current_player_move.mark)
+        end
+
+    end
+
+    describe '#check_if_win' do
+        it 'Should return true first horizontal line [X],[X],[X]' do 
```

Comment:
Use [Shared Examples](https://medium.freecodecamp.org/how-to-dry-out-your-rspec-tests-using-shared-examples-d5cc5d33fd76) here as well.